Support hashing with idtimestamp committed

robinbryce opened the pull request:
AB#8859

Commit:
Support hashing with idtimestamp committed

AB#8859

honourfish commented on simplehash/schemav2.go:
```diff
@@ -38,10 +39,20 @@ type HashOptions struct {
 
+// WithIDCommitted includes the snowflakeid unique commitment timestamp in the hash
+// idcommitted is never (legitimately) zero
+func WithIDCommitted(idcommitted uint64) HashOption {
```
We need to bump schema version for adding a new field?

I think i would prefer we have 2 schema versions, 1 with the `idcomitted` required and 1 without `idcommitted`.

Having optionality within the schema version may cause a headache down the line, i.e it could be one of these x hashes...

robinbryce replied:
I don't think that is necessary. This does not change the schema for simplehash. It is an accomodation for forestrie hashing. The idcommitted value is not explicitly in the event data at all

robinbryce replied:
as discussed, moved the options to a new file options.go to avoid ambiguity about what the schema is

Commit:
Move the options to their own implementation file

The options for controling the hasher are not part of the api event
schema. Putting them in their own file makes it clear their use is the
responsibility of the caller rather than formally part of the event
schema.
